### ISSUE TITLE
Query paramaters don't work for inserts

### ISSUE THREAD

**Issue by filimonov** [ClickHouse/ClickHouse#7174]
```
clickhouse-client --param_foo=bar --param_answer=42 

ClickHouse client version 19.15.1.1374 (official build).
Connecting to localhost:9000 as user default.
Connected to ClickHouse server version 19.15.1 revision 54426.

select {foo:String}, {answer:UInt8};

┌─CAST('bar', 'String')─┬─CAST(42, 'UInt8')─┐
│ bar                   │                42 │
└───────────────────────┴───────────────────┘

create table i (bar String, answer UInt8) Engine=Log;
Ok.

INSERT INTO i SELECT {foo:String}, {answer:UInt8};
Ok.

0 rows in set. Elapsed: 0.002 sec. 

INSERT INTO i VALUES ({foo:String}, {answer:UInt8});

Exception on client:
Code: 456. DB::Exception: Query parameter `foo` was not set

```

**Comment by filimonov** [comment]
Same for http:
```
echo '({foo:String}, {answer:UInt8})' | curl 'http://localhost:8123/?query=INSERT%20INTO%20i%20VALUES&param_foo=bar&param_answer=42' --data-binary @-
Code: 456, e.displayText() = DB::Exception: Query parameter `foo` was not set (version 19.15.1.1310 (official build))

echo 'SELECT {foo:String}, {answer:UInt8};' | curl 'http://localhost:8123/?param_foo=bar&param_answer=42' --data-binary @-
bar	42
```

**Comment by filimonov** [comment]
#5331 #5630

**Comment by filimonov** [comment]
Fixed in #6781. Thanks @tavplubix!
